src/args.rs

```diff
@@ -57,6 +57,18 @@ impl From<DeviceModel> for nitrokey::Model {
   }
 }
 
+impl<'de> serde::Deserialize<'de> for DeviceModel {
+  fn deserialize<D>(deserializer: D) -> Result<DeviceModel, D::Error>
+  where
+    D: serde::Deserializer<'de>,
+  {
+    use serde::de::Error;
+
+    let s = String::deserialize(deserializer)?;
+    s.parse().map_err(D::Error::custom)
```

Comment:
Could you make that `DeviceModel::from_str` please? It's way easier to understand what is happening this way, I'd say.